### ISSUE TITLE
Misleading Error message when Too many connections are built

### ISSUE THREAD

**Issue by Geigerkind** [blackbeam/rust-mysql-simple#193]
Hio!
During testing of my application I noticed that I was getting the following error message:

> thread 'account::tests::create::tests::password_empty' panicked at 'called `Result::unwrap()` on an `Err` value: DriverError { Packet out of sync }', src/libcore/result.rs:1165:5

The reason for it, during testing my applications opens as many mysql connections as there are threads and I assume that these connections are not properly closed after the test has finished.

So after executing my tests a few times I get this error message for some random tests.

Taking a look at my DB log files, I get this message:

> 2019-10-31 11:38:45 0 [Warning] Aborted connection 0 to db: 'unconnected' user: 'unauthenticated' host: 'connecting host' (Too many connections)

When increasing the maximum connections using:
`SET GLOBAL max_connections = 999999;`

The error is gone.

So your error message should read something along these lines.

Cheers!

**Comment by blackbeam** [comment]
BTW, had you updated to v17.0.0?

**Comment by Geigerkind** [comment]
Hmm, not sure if I can come up with code but here you go:
MySQL configuration: I used a ubuntu docker container which has a default configuration of 100 connections.
The application: These are just a bunch of test functions that are executed in parallel. Each of them connects to the database:
The code of the connection object that is spawned looks like this:
`pub struct MySQLConnection {
  pub con: mysql::Pool
}

impl MySQLConnection {
  pub fn new(db_name: &str) -> Self
  {
    dotenv().ok();
    let mut dns: String = env::var("MYSQL_DNS").unwrap();
    dns.push_str(db_name);
    MySQLConnection {
      con: mysql::Pool::new(dns).unwrap()
    }
  }
}`

A sample test looks like this:
`#[test]
  fn get_does_not_exist() {
    let account = Account::default();
    let acc_info = account.get(42);
    assert!(acc_info.is_err());
  }`

Note: 
`let account = Account::default();`
has a MySQLConnection object in it and creates a connection by default.

I am not closing the connection explicitly somewhere. I assume that the connection is dropped once the object goes out of scope.

Hope that helps!
Cheers!

**Comment by blackbeam** [comment]
`v17.0.0` is now published

**Comment by Geigerkind** [comment]
@blackbeam 
Sorry for the late reply, had to do lots for university.

The version update did not change the problem.

Cheers!

**Comment by blackbeam** [comment]
Couldn't reproduce, so closing for now. Feel free to reopen.
